File: Entities/Concrete/Product.cs
using System.Collections.Generic;
using Core.Entities;

namespace Entities.Concrete
{
    public class Product: IEntity
    {
        public int Id { get; set; }

        public string Name { get; set; }
        public decimal UnitPrice { get; set; }
        public int UnitsInStock { get; set; }

        public ICollection<Color> Colors { get; set; } = new List<Color>();
    }
    
}

File: UiPreparation/UI/src/app/components/app/models/product.ts
export class Product {
    id: number;
    name: string;
    unitPrice: number;
    unitsInStock: number;
    colors: { colorName: string }[];
    constructor(id: number, name: string, unitPrice: number, unitsInStock: number, colors: any ) {
        this.id = id;
        this.name = name;
        this.unitPrice = unitPrice;
        this.unitsInStock = unitsInStock;
        this.colors = colors;
    }
}
